resource/modules/resource_match.cpp

```diff
@@ -1541,32 +1541,34 @@ static int parse_R (std::shared_ptr<resource_ctx_t> &ctx,
         errno = EINVAL;
         goto out;
     }
-    if ((rc = json_unpack (o,
-                           "{s:i s:{s:I s:I} s?:o}",
-                           "version",
-                           &version,
-                           "execution",
-                           "starttime",
-                           &st,
-                           "expiration",
-                           &et,
-                           "scheduling",
-                           &graph))
+    if ((rc = json_unpack_ex (o,
+                              &error,
+                              0,
+                              "{s:i s:{s:F s:F} s?:o}",
+                              "version",
+                              &version,
+                              "execution",
+                              "starttime",
+                              &tstart,
+                              "expiration",
+                              &expiration,
+                              "scheduling",
+                              &graph))
         < 0) {
         errno = EINVAL;
-        flux_log (ctx->h, LOG_ERR, "%s: json_unpack", __FUNCTION__);
+        flux_log (ctx->h, LOG_ERR, "%s: json_unpack: %s", __FUNCTION__, error.text);
         goto freemem_out;
     }
-    if (version != 1 || st < 0 || et < st) {
+    if (version != 1 || tstart < 0 || expiration < tstart) {
```

Comment:
There needs to be an additional check for numeric limits:

```suggestion
    if (version != 1 || tstart < 0 || expiration < tstart || expiration > static_cast<double> (std::numeric_limits<int64_t>::max ()) {
```

Reply:
@garlick, if you're good with this I think this is otherwise GTG.

Reply:
OK by me!

Note I cannot approve this since I originally proposed it.  I'll leave it to you two to approve and set merge-when-passing when it's ready.